dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/registry/ServerNodeManager.java

```diff
@@ -201,13 +203,25 @@ public void notify(Event event) {
         }
 
         private void syncSingleWorkerNodeInfo(String workerAddress, WorkerHeartBeat info) {
+            if (!NetUtils.isLegalAddress(workerAddress)) {
```

Comment:
I don't think this is needed.

Reply:
> I don't think this is needed. You don't need to check all things.

This step is not overly precautionary. Because registry Event will contain intermediate nodes, such as `worker`. Please check the following screenshot.
![image](https://user-images.githubusercontent.com/20518339/204290503-d29213b5-72b3-4c1d-9cbb-2707a31e8d56.png)

Reply:
The problem is in registry, we need to fix the data in registry, this is a history data, right? 

It's not better to update the core class frequently unless serious bugs exist, every new change  may import new bug.

Reply:
> The problem is in registry, we need to fix the data in registry, this is a history data, right?
> 
> It's not a good idea to update the core class frequently unless serious bugs exist, every new change may import new bug.

this is a history data, right? -> No, it's wrong monitor path.

It's not a good idea to update the core class frequently unless serious bugs exist, every new change may import new bug. -> ok, i will rollback this part.

Reply:
after i rollback this validator, i updated the worker path check function, just exclude the `/node/worker`. becasue registry may raise `{path:/node/worker, data:null}` event.